examples/mqtt-dashboard/device/main.c

```diff
@@ -12,13 +12,13 @@ static void signal_handler(int signo) {
 // Mocked device pins
 static bool s_pins[100];
 
-void gpio_write(int pin, bool status) {
+void hal_gpio_write(int pin, bool status) {
```

Comment:
unrelated ground here.
We are removing these, they will be in hal.h, and there are some fixes too. #2516
Hal functions already have the name we are using.
Do you propose to have these in main and they call the hal ?
I'd rather have also a hal here, this code might run on a Raspberry Pi and light an actual LED or flip a relay.
We'll also keep main.c clean